packages/api/src/controllers/IdentitiesController.js

```diff
@@ -16,22 +19,55 @@ class IdentitiesController {
       return response.status(404).send({ message: 'not found' })
     }
 
+    const validatedAliases = await validateAliases(identity.aliases, identity.identifier, this.dapi)
+
     const balance = await this.dapi.getIdentityBalance(identifier)
 
-    response.send({ ...identity, balance })
+    response.send(Identity.fromObject({ ...identity, aliases: validatedAliases, balance }))
   }
 
   getIdentityByDPNS = async (request, response) => {
     const { dpns } = request.query
 
-    const identity = await this.identitiesDAO.getIdentityByDPNS(dpns)
+    let preIdentity
+    let identity
+
+    if (!dpns.includes('.')) {
```

Comment:
Very complex logic. what does this do?

Reply:
This logic is needed to get valid alias data from dapi, but if we don't specify the domain of the name, we can't request data from dapi, so we request data from indexer and then use it to get `contestedState`